proxy/http/HttpSM.cc

```diff
@@ -4879,7 +4867,7 @@ HttpSM::do_http_server_open(bool raw)
   // session when we already have an attached server session.
   else if ((TS_SERVER_SESSION_SHARING_MATCH_NONE == t_state.txn_conf->server_session_sharing_match || is_private()) &&
            (ua_txn != nullptr)) {
-    HttpServerSession *existing_ss = ua_txn->get_server_session();
+    Http1ServerSession *existing_ss = ua_txn->get_server_session();
```

Comment:
Http1?

Reply:
This forward-looking, the intent was to distinguish http1 from http2 and http3, which will come in later updates. It's easier to understand when we number all of them.

Reply:
I'm fine with the naming but wasn't sure why we use a concrete class here. So I thought it might be changed accidentally.

Does this mean everything inside HttpSM is H1? In other words, are we going to convert H2 and H3 transactions into H1 if we use HttpSM as their backends? I know now we do the conversion, however, I'm not too sure if we really be able to convert transactions like H2(client) -> H1 (HttpSM) -> H2(origin). We may need to handle transactions in a version independent flexible way in HttpSM. This is why I wondered why Http1.

Anyways, I understand this PR is for cleanup. I'm fine with Http1 if it's intentional.